Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -4711,40 +4777,127 @@ def panorama_query_logs_command(args: dict):
 
-    result = panorama_query_logs(log_type, number_of_logs, query, address_src, address_dst, ip_,
-                                 zone_src, zone_dst, time_generated, action,
-                                 port_dst, rule, url, filedigest)
+    if not job_id:
+        if query and (address_src or address_dst or zone_src or zone_dst
+                    or time_generated or action or port_dst or rule or url or filedigest):
+            raise Exception('Use the free query argument or the fixed search parameters arguments to build your query.')
 
-    if result['response']['@status'] == 'error':
-        if 'msg' in result['response'] and 'line' in result['response']['msg']:
-            raise Exception(f"Query logs failed. Reason is: {result['response']['msg']['line']}")
-        else:
-            raise Exception('Query logs failed.')
+        res = panorama_query_logs(
+            log_type, number_of_logs, query, address_src, address_dst, ip_,
+            zone_src, zone_dst, time_generated, action,
+            port_dst, rule, url, filedigest
+        )
+        result = ResponseObject(res)
 
-    if 'response' not in result or 'result' not in result['response'] or 'job' not in result['response']['result']:
-        raise Exception('Missing JobID in response.')
+        if result.response.status == 'error':
+            if result.response.msg:
+                raise Exception(f"Query logs failed. Reason is: {result.response.msg}")
+            else:
+                raise Exception('Query logs failed.')
 
-    query_logs_output = {
-        'JobID': result['response']['result']['job'],
-        'Status': 'Pending',
-        'LogType': log_type,
-        'Message': result['response']['result']['msg']['line']
-    }
+        if not result.response.result.job.id:
+            raise Exception('Missing JobID in response.')
+        query_logs_output = {
+            'JobID': result.response.result.job.id,
+            'Status': 'Pending',
+            'LogType': log_type,
+            'Message': result.response.msg
+        }
 
-    return_results({
-        'Type': entryTypes['note'],
-        'ContentsFormat': formats['json'],
-        'Contents': result,
-        'ReadableContentsFormat': formats['markdown'],
-        'HumanReadable': tableToMarkdown('Query Logs:', query_logs_output, ['JobID', 'Status'], removeNull=True),
-        'EntryContext': {"Panorama.Monitor(val.JobID == obj.JobID)": query_logs_output}
-    })
+        command_results = CommandResults(
+            outputs_prefix='Panorama.Monitor',
+            outputs_key_field='JobID',
+            outputs=query_logs_output,
+            readable_output=tableToMarkdown('Query Logs:',query_logs_output, ['JobID', 'Status'], removeNull=True),
+            raw_response=result.response.result.job.result
+        )
+
+        poll_result = PollResult(
+            response=command_results,
+            continue_to_poll=argToBoolean(args.get('polling', 'false')),
```

Comment:
This would then over-rule to users selection whether to use polling or not. Whether to continue to poll is then decided by the user selecting polling=True.

Reply:
that's not true, what will over-rule to users whether to use polling is the `polling` argument when passing it in the `args_for_next_run` which is something you have done. 

see my previous comment